File: neon.c
#include <stdio.h>

int isNeon(int num) {
  int square = num * num;
  int sum = 0;
  while (square > 0) {
    sum += square % 10;
    square /= 10;
  }
  return sum == num;
}

int main() {
  int num;
  printf("Enter a number: ");
  scanf("%d", &num);
  if (isNeon(num)) {
    printf("%d is a neon number.\n", num);
  } else {
    printf("%d is not a neon number.\n", num);
  }
  return 0;
}

File: neon.js
const isNeon = (num) => {
  let square = num * num;
  let sum = 0;
  while (square > 0) {
    sum += square % 10;
    square = Math.floor(square / 10);
  }
  return sum === num;
};

const num = parseInt(prompt("Enter a number:"));
if (isNeon(num)) {
  console.log(num + " is a neon number.");
} else {
  console.log(num + " is not a neon number.");
}
